R/rc_initialise.R

```diff
@@ -1,38 +1,363 @@
 #'
 #' @export
-rc_initialise <- function(B_LU_BRP,soil_properties){
+rc_initialise <- function(crops = NULL,
+                          amendment = NULL,
+                          dt.soc,
+                          rothc.parms,
+                          rothc.event,
+                          dt.time = NULL,
+                          start_date = NULL,
+                          soil_properties = NULL,
+                          dt.weather = NULL,
+                          initialization_method ='spinup_analytical_bodemcoolstof'){
   
   # add visual bindings
-  . = CIOM = CDPM = CRPM = CBIO = A_SOM_LOI = NULL
+  . = CIOM = CDPM = CRPM = CBIO = B_LU_EOM = M_CROPRESIDUE = B_LU_HC = chum.ini = NULL
+  P_DOSE = P_OM = M_GREEN_TIMING = fr_dpm_rpm = P_HC = B_LU_EOM_RESIDUE = NULL
+  abc = bd = time = toc = var = cf_abc = ciom.ini = biohum.ini = cbio.ini = NULL
+  A_SOM_LOI = B_C_OF_INPUT = P_C_OF = NULL
+
+  # Input validation by type
+  if (initialization_method == 'spinup_simulation') {
+    if (is.null(start_date)) stop("start_date is required for spinup_simulation")
+    if (is.null(crops)) stop("crops is required for spinup_simulation")
+    if (is.null(soil_properties)) stop("soil_properties is required for spinup_simulation")
+  }
+  
+  if (initialization_method %in% c('spinup_analytical_heuvelink','spinup_analytical_bodemcoolstof')) {
+    if (missing(dt.time) || is.null(dt.time)) stop("dt.time is required for analytical spin-up types")
+    if (missing(dt.soc) || is.null(dt.soc)) stop("dt.soc is required for analytical spin-up types")
+    if (is.null(dt.soc$A_CLAY_MI)) stop("dt.soc must contain A_CLAY_MI column")
+    if (is.null(dt.soc$toc)) stop("dt.soc must contain toc column")
+    }
+  
+  
+  # Define rothc parameters
+  k1 <- rothc.parms$k1
+  k2 <- rothc.parms$k2
+  k3 <- rothc.parms$k3
+  k4 <- rothc.parms$k4
+  
+  abc <- rothc.parms$abc
+  
+  # initialise options
+  
+  ## do a simulation for 150 years to estimate the C fractions assuming system is in equilibrium
+  if(initialization_method =='spinup_simulation'){
+   
+   # Extend crop and amendment files to include entire 150 year simulation
+    crop_extend <- if(!is.null(crops)){
+      rc_extend_crops(start_date = start_date, simyears = 150, crops = crops)
+    }else{
+      NULL
+    }
+    
+    amendment_extend <- if(!is.null(amendment)){
+      rc_extend_amendments(start_date = start_date, simyears = 150, amendments = amendment)
+    }else{
+      NULL
+    }
+ 
+    
+    # Set model parameters
+    parms <- list(unit = 'psomperfraction',
+                  initialization_method = 'none')
+    
+    # Set newly required inputs
+  
+    # Run initialization run for 150 years
+    this.result <- rc_sim(rothc_rotation = crop_extend,
+                          rothc_amendment = amendment_extend,
+                          soil_properties = soil_properties,
+                          weather = dt.weather,
+                          rothc_parms = parms)
+    
```

Comment:
seems ok, but by setting rothc_parms = parms you are using the default k1 to k4 values and abc. 
and do you need a check that soil_properties had only one option (or is that checked within rc_sim?)